aptos-move/framework/src/natives/aggregator_natives/context.rs

```diff
@@ -107,10 +113,22 @@ impl<'a> NativeAggregatorContext<'a> {
         }
 
         let delayed_field_changes = delayed_field_data.into_inner().into();
-
+        let delayed_write_set_keys = delayed_field_changes
+            .keys()
+            .cloned()
+            .collect::<HashSet<_>>();
         AggregatorChangeSet {
             aggregator_v1_changes,
-            delayed_field_changes: HashMap::from_iter(delayed_field_changes),
+            delayed_field_changes,
+            // is_empty check covers both whether delayed fields are enabled or not, as well as whether there
+            // are any changes that would require computing reads needing exchange.
+            // TODO[agg_v2](optimize) we only later compute the the write set, so cannot pass the correct skip values here.
+            reads_needing_exchange: if delayed_write_set_keys.is_empty() {
```

Comment:
It seems like we need the change set to be something like an enum with `Materialized`/`NonMaterialized` variants (if we wish to keep the same types in commit logic). Not for this PR, but curious what you think @igor-aptos ?

Reply:
you mean instead of having separate reads_needing_exchange, to have it be part of the regular write set? not sure that would be cleaner for all the usages we have